modules/assim.sequential/R/downscale_function.R

```diff
@@ -1,33 +1,90 @@
+##' @title Preprocess Data for Downscaling
+##' @name NA_preprocess
+##' @author Sambhav Dixit
+##'
+##' @param data_path Character. File path for .rds containing ensemble data.
+##' @param coords_path Character. File path for .csv file containing the site coordinates, with columns named "lon" and "lat".
+##' @param date Character. If SDA site run, format is yyyy/mm/dd; if NEON, yyyy-mm-dd. Restricted to years within the file supplied to 'data_path'.
```

Comment:
I think it is preferable for the function to take standard (ISO 8601) format YYYY-MM-DD and then adjust this internally ... in this case where both are in the same ymd order with different separators, they can be standardized in one line with 

```
newdate <- lubridate::hms(date)
if(is.na(newdate)) logger.error(paste("lubridate ymd function failed to parse", date))
```

or maybe we want to use tryCatch?

```
date <- tryCatch(
  lubridate::hms(date),
  condition = function(c) { 
    # condition catches warnings and errors
    logger.error(paste("lubridate hms function encountered an issue when parsing", date))
)
```

Reply:
now I see that the 'yyyy/mm/dd' format is required to access the correct part of the input_data. My inclination would be to convert both the date argument and the values of names(input_data) to the standard, thus:

```
names(input_data) <- lubridate::ymd(names(input_data))
``` 

If input_data has names that aren't dates, we may not want to overwrite them, which could be handled thus:

```
input_date_names <- suppressWarnings(as.character(lubridate::ymd(names(input_data))))
names(input_data) <- ifelse(is.na(input_date_names), 
                            names(input_data),
                            input_date_names)
```

Reply:
this line of suggestions to make the format more robust and flexible using `lubridate` has been addressed in the commit [5ce2339](https://github.com/PecanProject/pecan/pull/3321/commits/5ce2339328dec718109f18873c20f936af421fd6)

Reply:
If argument is now a Date, function documentation needs to be updated to reflect that